WooCommerce/Classes/POS/Card Present Payments/CardPresentPaymentOnboardingAdaptor.swift

```diff
@@ -0,0 +1,72 @@
+import Foundation
+import Combine
+import protocol Yosemite.StoresManager
+
+/// This is really a re-implementation of the CardPresentPaymentsOnboardingPresenter, as it needs to take the calls to `showOnboardingIfRequired` and
+/// route the output to a SwiftUI view for display, rather than directly displaying on the viewController that's passed in.
+final class CardPresentPaymentsOnboardingPresenterAdaptor: CardPresentPaymentsOnboardingPresenting {
+    private let stores: StoresManager
+
+    private let onboardingUseCase: CardPresentPaymentsOnboardingUseCase
+
+    private let readinessUseCase: CardPresentPaymentsReadinessUseCase
+
+    private let onboardingViewModel: CardPresentPaymentsOnboardingViewModel
+
+    private var readinessSubscription: AnyCancellable?
+
+    private(set) var onboardingScreenViewModelPublisher: AnyPublisher<CardPresentPaymentOnboardingPresentationEvent, Never>
```

Comment:
super nit: if we don't plan to change the publisher
```suggestion
    let onboardingScreenViewModelPublisher: AnyPublisher<CardPresentPaymentOnboardingPresentationEvent, Never>
```